test/test-with-server.js

```diff
@@ -235,6 +236,21 @@ describe("follow-redirects", function () {
       });
   });
 
+  it("should callback setTimeout on the returned stream", function () {
+    app.get("/redirect", redirectsTo("/timeout"));
+    app.get("/timeout", delaysJson(5000, { time: "out" }));
+
+    return server.start(app)
+      .then(asPromise(function (resolve, reject) {
+        var req = http.get("http://localhost:3600/redirect", reject);
+        req.on("error", reject);
+        req.setTimeout(10, function () {
+          req.abort();
```

Comment:
But what are we proving if we abort ourselves?

Reply:
I don't think the request is actually aborted by the timeout - you have to do that yourself.  So the test suite hangs without this (until it hits the 2000ms limit).

Reply:
Tried this in node 10.7.0 ...

```js
const http = require('http');
const url = 'http://slowwly.robertomurray.co.uk/delay/5000/url/http://www.google.com';

console.log(`GET slow url ->`);
const start = new Date().getTime();
const timer = () => (new Date().getTime() - start) + 'ms';

const req = http.get(url, res => {
  console.log(`  got ${res.statusCode} (${timer()})`);
});
req.on('error', err => {
  console.log(`  got error (${timer()})`, err.message);
});
req.setTimeout(1000, () => {
  // req.abort();
  console.log(`  timeout callback (${timer()})`);
});
process.on('exit', () => {
  console.log(`  exiting (${timer()})`)
});
```

Which prints (with abort commented out vs in):

```
// no abort
GET slow url ->
  timeout callback (1167ms)
  got 302 (5312ms)
  exiting (5316ms)

// with abort()
GET slow url ->
  timeout callback (1185ms)
  got error (1187ms) socket hang up
  exiting (1189ms)
```